algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py

```diff
@@ -140,27 +180,51 @@ def loss_fn(self, label_batch: spec.Tensor,
     one_hot_targets = jax.nn.one_hot(label_batch, 10)
     return -jnp.sum(one_hot_targets * nn.log_softmax(logits_batch), axis=-1)
 
-  @functools.partial(
-      jax.pmap,
-      axis_name='batch',
-      in_axes=(None, 0, 0, 0, None),
-      static_broadcasted_argnums=(0,))
-  def _eval_model(
-      self,
-      params: spec.ParameterContainer,
-      batch: Dict[str, spec.Tensor],
-      model_state: spec.ModelAuxiliaryState,
-      rng: spec.RandomState) -> Tuple[spec.Tensor, spec.ModelAuxiliaryState]:
-    logits, _ = self.model_fn(
-        params,
-        batch,
-        model_state,
-        spec.ForwardPassMode.EVAL,
-        rng,
-        update_batch_norm=False)
-    accuracy = jnp.sum(jnp.argmax(logits, axis=-1) == batch['targets'])
-    loss = jnp.sum(self.loss_fn(batch['targets'], logits))
-    num_data = len(logits)
-    metrics = {'accuracy': accuracy, 'loss': loss, 'num_data': num_data}
+  def _compute_metrics(self, logits, labels):
+    loss = jnp.sum(self.loss_fn(labels, logits))
+    # not accuracy, but nr. of correct predictions
+    accuracy = jnp.sum(jnp.argmax(logits, -1) == labels)
+    metrics = {
+        'loss': loss,
+        'accuracy': accuracy,
+    }
     metrics = lax.psum(metrics, axis_name='batch')
     return metrics
+
+  def _eval_model_on_split(self,
+                           split: str,
+                           num_examples: int,
+                           global_batch_size: int,
+                           params: spec.ParameterContainer,
+                           model_state: spec.ModelAuxiliaryState,
+                           rng: spec.RandomState,
+                           data_dir: str):
+    data_rng, model_rng = jax.random.split(rng, 2)
+    # Sync batch statistics across replicas before evaluating.
+    model_state = self.sync_batch_stats(model_state)
+    num_batches = int(math.ceil(num_examples / global_batch_size))
+    # We already repeat the dataset indefinitely in tf.data.
+    if split not in self._eval_iters:
+      eval_iter = self.build_input_queue(
+          data_rng,
+          split=split,
+          global_batch_size=global_batch_size,
+          data_dir=data_dir,
+          cache=True,
+          repeat_final_dataset=True,
+          num_batches=num_batches)
+      # Note that this stores the entire eval dataset in memory.
+      self._eval_iters[split] = itertools.cycle(eval_iter)
```

Comment:
This is not really necessary, since `repeat_final_dataset=True`, right? Unless we actually want to keep the eval sets in memory; however, this is not the case for the PyTorch version. Also, won't the memory consumption not increase with every evaluation, since the iterator never hits `StopIteration`, i.e. the data which is stored in memory is actually not used for the next evaluation?

Reply:
That is correct -- I removed this line.